### ISSUE TITLE
Upload to Google Drive

### ISSUE THREAD

**Issue by orschiro** [lupoDharkael/flameshot#608]
I am concerned of using Imgur hosting for privacy sensitive screenshots.

Can we add an option to upload to Google Drive instead?

Has better privacy and sharing settings than Imgur. 

Thanks!

**Comment by AlexP11223** [comment]
You can simply save images to the Google Dtive directory. 

And then use something else (e.g. in Insync client and there is integration with file managers) to copy the link or open in web browser.

A shameless plug: my simple script to convert Google Drive standard sharing links into direct links. https://github.com/AlexP11223/gdrive_link_transformer

**Comment by noraj** [comment]
For Windows there is [ShareX](https://getsharex.com/) but I don't think that flameshot should support more than Imgur. Else it will be an endless `Feature request - XXX service upload` and FlameShot is a screenshot utility not an upload manager.

**Comment by orschiro** [comment]
Agree but then there should be the option to choose whether to upload to
Imgur private or public.

**Comment by AlexP11223** [comment]
Yes, and this can be done by adding support for user API keys and OAuth auth.

Someone started started implementing this some time ago, but then left :( https://github.com/lupoDharkael/flameshot/pull/393

**Comment by konradlu** [comment]
for example, the screencloud application has the ability to add plugins for various services
https://github.com/search?p=2&q=screencloud+plugin&type=Repositories

but screencloud has worse options than flameshot

**Comment by fabianski7** [comment]
if you care about privacy, you should go away from google services

**Comment by borgmanJeremy** [comment]
Please track this RFC for future discussion on this topic: https://github.com/flameshot-org/flameshot/pull/955
